modules/heat_conduction/src/bcs/GaussianWeldEnergyFluxBC.C

```diff
@@ -0,0 +1,49 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GaussianWeldEnergyFluxBC.h"
+#include "Function.h"
+
+registerMooseObject("HeatConductionApp", GaussianWeldEnergyFluxBC);
+
+InputParameters
+GaussianWeldEnergyFluxBC::validParams()
+{
+  InputParameters params = ADIntegratedBC::validParams();
+  params.addRequiredParam<Real>("reff",
+                                "The effective radius describing the radial distribution of the "
+                                "beam energy. This should be non-dimensional.");
+  params.addRequiredParam<Real>("F0", "The average heat flux of the laser");
+  params.addRequiredParam<Real>("R", "The beam radius");
+  params.addParam<FunctionName>("x_beam_coord", 0, "The x coordinate of the center of the beam");
+  params.addParam<FunctionName>("y_beam_coord", 0, "The y coordinate of the center of the beam");
+  params.addParam<FunctionName>("z_beam_coord", 0, "The z coordinate of the center of the beam");
+  return params;
+}
+
+GaussianWeldEnergyFluxBC::GaussianWeldEnergyFluxBC(const InputParameters & params)
+  : ADIntegratedBC(params),
+    _reff(getParam<Real>("reff")),
+    _F0(getParam<Real>("F0")),
+    _R(getParam<Real>("R")),
+    _x_beam_coord(getFunction("x_beam_coord")),
+    _y_beam_coord(getFunction("y_beam_coord")),
+    _z_beam_coord(getFunction("z_beam_coord"))
+{
+}
+
+ADReal
+GaussianWeldEnergyFluxBC::computeQpResidual()
+{
+  RealVectorValue beam_coords{_x_beam_coord.value(_t, _q_point[_qp]),
+                              _y_beam_coord.value(_t, _q_point[_qp]),
+                              _z_beam_coord.value(_t, _q_point[_qp])};
+  auto r = (_ad_q_points[_qp] - beam_coords).norm();
+  return -_test[_i][_qp] * 2. * _reff * _F0 * std::exp(-_reff * r * r / (_R * _R));
```

Comment:
you re going to loose energy from integration error here. Unless you're planning to adapt _F 0?

Reply:
How would you combat that?

Reply:
I could just drop this PR and use some parsed function class

Reply:
> outside of MOOSE, approximating the Gaussian with some polynomial shape that is integrated exactly by the quadrature.

If not done with a lot of thought, this would introduce the exact kind of physics to modeling error

> refining the mesh locally near the laser spot

This is done in the input files using this BC in https://github.com/idaholab/moose/pull/24204 by refining for large gradient jumps (e.g. for large Laplacian) for the temperature field.

Reply:
There is `FunctionNeumannBC` which I can use in #24204. I don't want to spend a lot of time on this, so if we don't want this BC as written, then I will just move on and close this

Reply:
I think the BC is fine. 
I would remove "Weld" from the name though